gap/NautyGraph.gi

```diff
@@ -361,6 +361,40 @@ InstallMethod( CanonicalForm,
     
 end );
 
+InstallMethod( CanonicalForm,
+               [ IsNautyEdgeColoredGraphRep ],
+               
+  function( graph )
+    local colors, edges, perm, new_graph, permEdges, i,
+        permInv, edgeClass, newClass;
+    
+    perm := CanonicalLabeling( graph );
+    permInv := CanonicalLabelingInverse( graph );
+    
+    edges := graph!.edge_list;
+
+    permEdges := [];
+    # Since this computation should be efficient we do it manually
+    for edgeClass in edges do
+        newClass := [];
+
+        for i in [ 1 .. Length( edgeClass ) ] do # edges is a dense list
+            # Every line of code in this loop costs a lot of time
+            Add( newClass, [ edgeClass[ i ][ 1 ]^permInv, edgeClass[ i ][ 2 ]^permInv ] );
+            if newClass[ i ][ 1 ] > newClass[ i ][ 2 ] then
+                newClass[ i ] := [ newClass[ i ][ 2 ], newClass[ i ][ 1 ] ];
+            fi;
+        od;
```

Comment:
As mentioned, I would recommend also benchmarking this variant, as `OnSets` is a kernel function (which of course handles arbitrary sets, not just sets of size 2, so it is not as efficient as possible, but it still might beat the current code)
```suggestion
        for edge in edgeClass do # edges is a dense list
            # Every line of code in this loop costs a lot of time
            Add( newClass, OnSets( edge, permInv ) );
        od;
```
I would also try this:
```suggestion
        newClass := List( edgeClass, edge -> OnSets( edge, permInv ) );
```

Reply:
Even if `OnSets` turns out not to be faster, you definitely can improve the code by using local variables: i.e. something along these lines:
```gap
a:= edge[1]^permInv;
b:= edge[2]^permInv;
if a < b then
   Add(newClass, [a,b]);
else
   Add(newClass, [b,a]);
fi;
```